core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -128,7 +134,8 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
                 filterSplitResult.getPushableConditions(),
                 // TODO we could pass only subset of assignments here, those which are needed to resolve filterSplitResult.getPushableConditions
                 leftAssignments,
-                rightAssignments);
+                rightAssignments,
+                getJoinStatistics(joinNode, left, right, context));
```

Comment:
Here or at getJoinStatistics() definition, include the comment _why_ we're doing this, as this is not obvious. One could think connector could determine this on its own, at least for table scans.

The reasons are:

- the engine knows how to estimate join and connector may not
- the engine may have cached stats for the table scans (within `context.getStatsProvider()`), so can be able to provide information more inexpensively 
- in the future, the engine can be able to provide stats for table scan even in case when connector no longer can, linking to https://github.com/trinodb/trino/issues/6998

Reply:
(@losipiuk  edited, added fourth bullet)

Reply:
Shouldn't the comment be rather in SPI method javadoc?

Reply:
(I put it in rule for now)